hetmech/test_degree_weight.py

```diff
@@ -31,7 +33,117 @@ def test_disease_gene_example_dwwc():
+    # Dictionary with tuples of matrix and percent nonzero
+    mat_dict = {
+        'GiGaD': ([[0.25, 0.],
+                   [0.35355339, 0.],
+                   [0., 0.6767767],
+                   [0.35355339, 0.],
+                   [0., 0.35355339],
+                   [0., 0.],
+                   [0.35355339, 0.]], 0.429),
+        'GaDaG': ([[0.25, 0.25, 0., 0.25, 0., 0.1767767, 0.],
+                   [0.25, 0.25, 0., 0.25, 0., 0.1767767, 0.],
+                   [0., 0., 0.5, 0., 0., 0.35355339, 0.],
+                   [0.25, 0.25, 0., 0.25, 0., 0.1767767, 0.],
+                   [0., 0., 0., 0., 0., 0., 0.],
+                   [0.1767767, 0.1767767, 0.35355339, 0.1767767, 0., 0.375,
+                    0.],
+                   [0., 0., 0., 0., 0., 0., 0.]], 0.388),
+        'GeTlD': ([[0., 0.],
+                   [0., 0.],
+                   [0., 0.70710678],
+                   [0., 0.],
+                   [0., 0.],
+                   [0., 0.],
+                   [0., 0.]], 0.071),
+        'GiG': ([[0., 0., 0.35355339, 0., 0.70710678, 0., 0.],
+                 [0., 0., 0.5, 0., 0., 0., 0.],
+                 [0.35355339, 0.5, 0., 0.5, 0., 0., 0.5],
+                 [0., 0., 0.5, 0., 0., 0., 0.],
+                 [0.70710678, 0., 0., 0., 0., 0., 0.],
+                 [0., 0., 0., 0., 0., 0., 0.],
+                 [0., 0., 0.5, 0., 0., 0., 0.]], 0.204),
+        'GaDaG_dwpc': ([[0., 0.25, 0., 0.25, 0., 0.1767767, 0.],
+                        [0.25, 0., 0., 0.25, 0., 0.1767767, 0.],
+                        [0., 0., 0., 0., 0., 0.35355339, 0.],
+                        [0.25, 0.25, 0., 0., 0., 0.1767767, 0.],
+                        [0., 0., 0., 0., 0., 0., 0.],
+                        [0.1767767, 0.1767767, 0.35355339, 0.1767767, 0.,
+                         0., 0.],
+                        [0., 0., 0., 0., 0., 0., 0.]], 0.288)
+    }
+    node_dict = {
+        'G': ['CXCR4', 'IL2RA', 'IRF1', 'IRF8', 'ITCH', 'STAT3', 'SUMO1'],
+        'D': ["Crohn's Disease", 'Multiple Sclerosis'],
+        'T': ['Leukocyte', 'Lung']
+    }
+    type_dict = {
+        True: numpy.array,
+        False: sparse.csc_matrix
+    }
+    row_names = node_dict[m_path[0]]
+    col_names = node_dict[m_path[-1]]
+
+    if m_path == 'GaDaG':
+        dwwc_mat = mat_dict['GaDaG'][0]
+        dwpc_mat = mat_dict['GaDaG_dwpc'][0]
+        type_dwwc = type_dict[mat_dict['GaDaG'][1] >= threshold]
+        type_dwpc = type_dict[mat_dict['GaDaG_dwpc'][1] >= threshold]
+    else:
+        dwwc_mat = dwpc_mat = mat_dict[m_path][0]
+        type_dwwc = type_dict[mat_dict[m_path][1] >= threshold]
+        type_dwpc = type_dict[mat_dict[m_path][1] >= threshold]
+
+    dwwc_mat = type_dwwc(dwwc_mat)
+    dwpc_mat = type_dwpc(dwpc_mat)
+
+    return type_dwwc, dwwc_mat, type_dwpc, dwpc_mat, row_names, col_names
+
+
+@pytest.mark.parametrize('thresh', (0, 0.25, 0.3, 0.5, 0.7, 1))
+@pytest.mark.parametrize('m_path', ('GiGaD', 'GaDaG', 'GeTlD', 'GiG'))
```

Comment:
Will leave `m_path` here to distinguish the metapath as a string (`m_path`) from the metapath object which will be-- in the test function-- `metapath`.